php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -0,0 +1,1538 @@
+        MULTIPLICATIVE_EXPR(),
+        b.zeroOrMore(f.newTuple62(
+          b.firstOf(b.token(PLUS), b.token(MINUS)),
+          MULTIPLICATIVE_EXPR()))));
+  }
+
+  public ExpressionTree SHIFT_EXPR() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.SHIFT_EXPR).is(
+      f.shiftExpr(
+        ADDITIVE_EXPR(),
+        b.zeroOrMore(f.newTuple63(
+          b.firstOf(b.token(SL), b.token(SR)),
+          ADDITIVE_EXPR()))));
+  }
+
+  public ExpressionTree RELATIONAL_EXPR() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.RELATIONAL_EXPR).is(
+      f.relationalExpr(
+        SHIFT_EXPR(),
+        b.zeroOrMore(f.newTuple64(
+          b.firstOf(b.token(LE), b.token(GE), b.token(LT), b.token(GT)),
+          SHIFT_EXPR()))));
+  }
+
+  public ExpressionTree EQUALITY_EXPR() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.EQUALITY_EXPR).is(
+      f.equalityExpr(
+        RELATIONAL_EXPR(),
+        b.zeroOrMore(f.newTuple65(
+          b.firstOf(b.token(NOTEQUAL2), b.token(NOTEQUAL), b.token(EQUAL2), b.token(EQUAL), b.token(NOTEQUALBIS)),
+          RELATIONAL_EXPR()))));
+  }
+
+  public ExpressionTree BITWISE_AND_EXPR() {
+    return b.<ExpressionTree>nonterminal(Kind.BITWISE_AND).is(
+      f.bitwiseAndExpr(
+        EQUALITY_EXPR(),
+        b.zeroOrMore(f.newTuple66(
+          b.token(PHPPunctuator.AMPERSAND),
+          EQUALITY_EXPR()))));
+  }
+
+  public ExpressionTree BITWISE_XOR_EXPR() {
+    return b.<ExpressionTree>nonterminal(Kind.BITWISE_XOR).is(
+      f.bitwiseXorExpr(
+        BITWISE_AND_EXPR(),
+        b.zeroOrMore(f.newTuple67(
+          b.token(PHPPunctuator.XOR),
+          BITWISE_AND_EXPR()))));
+  }
+
+  public ExpressionTree BITWISE_OR_EXPR() {
+    return b.<ExpressionTree>nonterminal(Kind.BITWISE_OR).is(
+      f.bitwiseOrExpr(
+        BITWISE_XOR_EXPR(),
+        b.zeroOrMore(f.newTuple68(
+          b.token(PHPPunctuator.OR),
+          BITWISE_XOR_EXPR()))));
+  }
+
+  public ExpressionTree CONDITIONAL_AND_EXPR() {
+    return b.<ExpressionTree>nonterminal(Kind.CONDITIONAL_AND).is(
+      f.logicalAndExpr(
+        BITWISE_OR_EXPR(),
+        b.zeroOrMore(f.newTuple69(
+          b.firstOf(b.token(PHPPunctuator.ANDAND), b.token(PHPKeyword.AND)),
+          BITWISE_OR_EXPR()))));
+  }
+
+  public ExpressionTree CONDITIONAL_XOR_EXPR() {
+    return b.<ExpressionTree>nonterminal(Kind.ALTERNATIVE_CONDITIONAL_XOR).is(
+      f.logicalXorExpr(
+        CONDITIONAL_AND_EXPR(),
+        b.zeroOrMore(f.newTuple70(
+          b.token(PHPKeyword.XOR),
+          CONDITIONAL_AND_EXPR()))));
+  }
+
+  public ExpressionTree CONDITIONAL_OR_EXPR() {
+    return b.<ExpressionTree>nonterminal(Kind.CONDITIONAL_OR).is(
+      f.logicalOrExpr(
+        CONDITIONAL_XOR_EXPR(),
+        b.zeroOrMore(f.newTuple71(
+          b.firstOf(b.token(PHPPunctuator.OROR), b.token(PHPKeyword.OR)),
+          CONDITIONAL_XOR_EXPR()))));
+  }
+
+  public ExpressionTree CONDITIONAL_EXPR() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00100)